### PR TITLE
Display catch return info if any salmon licences are purchased

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/IWTF-3316

This fixes a bug in multibuy where we were only displaying the catch return info on the order complete page if the most recently added licence was for salmon fishing.

This info should be displayed if any of the licences added are for salmon fishing, not just the latest one.

### CODE REVIEW

**Comment by irisfaraway** [comment]
Also I refactored a bunch of tests to reduce duplication (couldn't help myself)

**Comment by irisfaraway** [comment]
Do not review/merge for now – will revisit after https://github.com/DEFRA/rod-licensing/pull/1668 is done

**Comment by irisfaraway** [comment]
Now ready to review. Still includes the test refactor to try to simplify the tests and make them more consistent. Also removes some accidental cruft from the order-complete page.

**Review by jaucourt — CHANGES_REQUESTED** [review]

File: packages/gafl-webapp-service/src/pages/order-complete/order-complete/route.js [hunk 1]
```diff
@@ -7,10 +7,18 @@ import * as mappings from '../../../processors/mapping-constants.js'
 import { nextPage } from '../../../routes/next-page.js'
 import { addLanguageCodeToUri } from '../../../processors/uri-helper.js'
 
+const checkForSalmonPermits = transaction => {
+  for (const permission of transaction.permissions) {
+    if (permission.licenceType === mappings.LICENCE_TYPE['salmon-and-sea-trout']) {
+      return true
+    }
+  }
+  return false
```
[the remark this answers is not in the document]
Comment (irisfaraway) [reply packages/gafl-webapp-service/src/pages/order-complete/order-complete/route.js:16]: Very nice, I did not know that existed! Thanks 👍

**Review by jaucourt — APPROVED** [review]

File: packages/gafl-webapp-service/src/pages/order-complete/order-complete/__tests__/order-complete.spec.js [hunk 2]
```diff
@@ -2,89 +2,65 @@ import { LICENCE_DETAILS, NEW_TRANSACTION, ORDER_COMPLETE } from '../../../../ur
...
-jest.mock('../../../../processors/date-and-time-display.js')
 jest.mock('../../../../processors/uri-helper.js')
-jest.mock('../../../../constants.js', () => ({
-  ...jest.requireActual('../../../../constants.js'),
-  COMPLETION_STATUS: {
-    agreed: 'alright then',
-    posted: 'in the letterbox',
-    finalised: 'no going back now',
-    completed: 'all done'
-  },
```
Comment (jaucourt) [inline packages/gafl-webapp-service/src/pages/order-complete/order-complete/__tests__/order-complete.spec.js:18]: I've approved, but you've reduced the quality of the tests and reinforced the coupling between constants and the order complete getData function by removing this mocking. It's not a massive danger, as I doubt we're going to ever remove that, but that's why I did this.

**Pull request merged**